gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -133,8 +133,22 @@ private <T> T withRevWalkUnchecked(CheckedFunction1<RevWalk, T> fun) {
     }
   }
 
-  private boolean isBranchPresent(String branchFullName) {
-    return Try.of(() -> jgitRepoForMainGitDir.resolve(branchFullName)).getOrNull() != null;
+  private boolean isBranchPresent(String branchFullName, String currentlyCheckedBranchFullName) {
+    boolean branchExists = Try.of(() -> jgitRepoForMainGitDir.resolve(currentlyCheckedBranchFullName)).getOrNull() != null;
+    if (branchFullName.equals(currentlyCheckedBranchFullName)) {
+      return branchExists;
+    } else if (branchExists) {
+      return false;
+    } else {
+      @SuppressWarnings("index:argument") String remainingWithoutBeginningSlash = branchFullName
```

Comment:
Checker's warning here is completely justified, in principle there's no compile-time guarantee that `currentlyCheckedBranchFullName` is shorter than `branchFullName` (and hence, that `branchFullName.substring(currentlyCheckedBranchFullName.length() + 1)` will complete successfully)

Reply:
It's a different question whether it can be suppressed (since we know otherwise that `substring` call is correct)... let's assume for now that it can